source/simplex/quadrature_lib.cc

```diff
@@ -84,41 +84,9 @@ namespace Simplex
           }
         else if (n_points_1D == 4)
           {
-            const double q12 = 0.5;
-
-            // clang-format off
-            this->quadrature_points.emplace_back(0.06493051315916486, 0.06493051315916486);
-            this->quadrature_points.emplace_back(0.06493051315916486, 0.8701389736816703);
-            this->quadrature_points.emplace_back(0.8701389736816703, 0.06493051315916486);
-            this->quadrature_points.emplace_back(0.2845755842491703, 0.517039939069323);
-            this->quadrature_points.emplace_back(0.3135591843849315, 0.04386347179237249);
-            this->quadrature_points.emplace_back(0.19838447668150672, 0.2845755842491703);
-            this->quadrature_points.emplace_back(0.642577343822696, 0.3135591843849315);
-            this->quadrature_points.emplace_back(0.517039939069323, 0.19838447668150672);
-            this->quadrature_points.emplace_back(0.04386347179237249, 0.642577343822696);
-            this->quadrature_points.emplace_back(0.517039939069323, 0.2845755842491703);
-            this->quadrature_points.emplace_back(0.04386347179237249, 0.3135591843849315);
-            this->quadrature_points.emplace_back(0.19838447668150672, 0.517039939069323);
-            this->quadrature_points.emplace_back(0.642577343822696, 0.04386347179237249);
-            this->quadrature_points.emplace_back(0.2845755842491703, 0.19838447668150672);
-            this->quadrature_points.emplace_back(0.3135591843849315, 0.642577343822696);
-            // clang-format on
-
-            this->weights.emplace_back(q12 * 0.053077801790232484);
-            this->weights.emplace_back(q12 * 0.053077801790232484);
-            this->weights.emplace_back(q12 * 0.053077801790232484);
-            this->weights.emplace_back(q12 * 0.07085308369213365);
-            this->weights.emplace_back(q12 * 0.069274682079417);
-            this->weights.emplace_back(q12 * 0.07085308369213365);
-            this->weights.emplace_back(q12 * 0.069274682079417);
-            this->weights.emplace_back(q12 * 0.07085308369213365);
-            this->weights.emplace_back(q12 * 0.069274682079417);
-            this->weights.emplace_back(q12 * 0.07085308369213365);
-            this->weights.emplace_back(q12 * 0.069274682079417);
-            this->weights.emplace_back(q12 * 0.07085308369213365);
-            this->weights.emplace_back(q12 * 0.069274682079417);
-            this->weights.emplace_back(q12 * 0.07085308369213365);
-            this->weights.emplace_back(q12 * 0.069274682079417);
+            QWitherdenVincent<dim> quad(n_points_1D);
+            this->quadrature_points = quad.get_points();
+            this->weights           = quad.get_weights();
```

Comment:
I think you should be able to just write this as
```
  *this = QWitherdenVincent<dim>(n_points_1D);
```
But I don't quite understand. The documentation for `Simplex::QGauss` says that for n=4, you have 15 points, whereas QW...V... says it's 19 points.

Reply:
Good catch on the documentation. The copy constructor didn't work so I went ahead and did it manually - I think we need to add a copy constructor to `SimplexQuadrature`.

Reply:
Why doesn't it just use the copy constructor of the `Quadrature` base class?

Reply:
> But I don't quite understand. The documentation for Simplex::QGauss says that for n=4, you have 15 points, whereas QW...V... says it's 19 points.

I think you misread the `n = 5` case for QWV as `n = 4` - we have an extra rule here that's not yet in QGauss.

Reply:
The copy operator doesn't work this way. `Quadrature<dim>` defines a copy operator. However, each inheriting class defines its own implicit copy operator, which hides the explicitly-defined base class operator. I implemented the fix with `static_cast`.

Reply:
Can't you just to `this->Quadrature<dim>::operator=(rhs);`? You could even omit the `this->`, I just added it for clarity.

Reply:
Oh yeah, that's a much better idea.

Reply:
I implemented that fix.